test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
+    describe('Test Array methods ', () => {
+        it('Should annotate class', () => {
+            class MyClass {
+                @ArraySchema()
+                myProperty!: Array<number>
+            }
+            const metadata = Reflect.getMetadata(WORKING_SCHEMA_KEY, MyClass.prototype);
+            const expected = {
+                myProperty: Joi.array()
+            }
+            expect(metadata).toEqual(expected)
+        })
```

Comment:
⛏ This codebase uses semicolons to terminate all statements, please add any missing semicolons.

There's a few other formatting inconsistencies to address. PR #32 was recently merged in, adding linting rules. Please merge from `master` and run tslint. (You may be able to pass the `-f` flag to automatically fix any linting issues identified.)